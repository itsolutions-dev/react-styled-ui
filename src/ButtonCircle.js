import styled from 'styled-components';
import Button from './Button';

const ButtonCircle = styled(Button)`
  border-radius: 9999px;
`;

export default ButtonCircle;
