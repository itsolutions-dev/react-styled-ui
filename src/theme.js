export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
