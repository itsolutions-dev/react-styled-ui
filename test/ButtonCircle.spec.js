import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import ButtonCircle from '../src/ButtonCircle';

describe('ButtonCircle', () => {
  test('should render a ButtonCircle', () => {
    const component = ReactTestRenderer.create(<ButtonCircle theme={theme} />);
    expect(component).toBeDefined();
  });

  test('should have a button tag', () => {
    const component = ReactTestRenderer.create(<ButtonCircle theme={theme} />).toJSON();
    expect(component.type).toEqual('button');
  });

  test('should have a border-radius', () => {
    const component = ReactTestRenderer.create(
      <ButtonCircle
        theme={theme}
      />,
    );
    expect(component).toHaveStyleRule('border-radius', '9999px');
  });
});
