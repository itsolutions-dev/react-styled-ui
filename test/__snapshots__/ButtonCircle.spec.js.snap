// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonCircle should render a ButtonCircle 1`] = `
.c0 {
  padding: 8px 16px;
  margin-left: 5px;
  margin-right: 5px;
  font-size: 16px;
  background: #ffffff;
  color: #4c4c4c;
  opacity: 1;
  border: 2px solid #bababa;
  border-radius: 9999px;
}

.c0:hover {
  background: #f5f7f7;
}

.c0:focus {
  outline: none;
}

<button
  className="c0"
/>
`;
